vips/foreign.go

```diff
@@ -102,90 +71,145 @@ const (
 	TiffPredictorFloat      TiffPredictor = C.VIPS_FOREIGN_TIFF_PREDICTOR_FLOAT
 )
 
+var (
```

Comment:
you can merge the header into the ImageTypeInfo struct, including the indexes and have each struct to expose a "Me" function that does the match